File: app/src/main/java/fiveagency/internship/food/movieapp/ui/movieslist/MoviesListContract.java
package fiveagency.internship.food.movieapp.ui.movieslist;

import java.util.List;

import fiveagency.internship.food.domain.model.WeatherModel;

public interface MoviesListContract {

    interface Presenter {

        void start();

        void showMovieDetails(int movieId);

        void getFlowableMoviesUseCase();

        void setView(MoviesListContract.View view);

        void insertFavorite(int movieId);

        void removeFavorite(int movieId);

        void saveMovies(List<MovieViewModel> movieViewModelList);

        void getAdditionalMovies(int page);

        void onStop();

        void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults);

        void showSearchScreen();
    }

    interface View {

        void render(final MoviesListViewModel moviesListViewModel);

        void renderWeather(WeatherModel fiveDayForecast);
    }
}

File: app/src/main/java/fiveagency/internship/food/movieapp/ui/movieslist/MoviesListContract.kt
package fiveagency.internship.food.movieapp.ui.movieslist

import fiveagency.internship.food.domain.model.WeatherModel

interface MoviesListContract {
    interface Presenter {
        fun start()
        fun showMovieDetails(movieId: Int)
        fun getFlowableMovies()
        fun setView(view: View)
        fun insertFavorite(movieId: Int)
        fun removeFavorite(movieId: Int)
        fun saveMovies(movieViewModelList: List<MovieViewModel>)
        fun getAdditionalMovies(page: Int)
        fun onStop()
        fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray)
        fun showSearchScreen()
        fun getFlowableFavoriteRecommendations()
        fun getFlowableWeatherRecommendations()
        fun onRefresh()
    }

    interface View {
        fun render(moviesListViewModel: MoviesListViewModel)
        fun renderWeather(fiveDayForecast: WeatherModel)
    }
}